### PR TITLE
feat: add dsn_util

### PR DESCRIPTION
**Pull request by monserrat101**

### CODE REVIEW

File: pkg/dsn_util/postgresql.go [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+}
+
+type Option func(cfg *Config)
+
+// Deprecated. Use Option instead.
+type DriverOption = Option
+
+func WithNetwork(network string) Option {
+	if network == "" {
+		panic("network is empty")
```
Comment (BinLiam) [inline pkg/dsn_util/postgresql.go:126]: Oh, that's too bad.

File: pkg/dsn_util/postgresql.go [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+	}
+	return func(cfg *Config) {
+		cfg.Network = network
+	}
+}
+
+func WithAddr(addr string) Option {
+	if addr == "" {
```
Comment (BinLiam) [inline pkg/dsn_util/postgresql.go:134]: Oh, that's too bad.

File: pkg/dsn_util/postgresql.go [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+			cfg.TLSConfig = nil
+		} else {
+			cfg.TLSConfig = &tls.Config{InsecureSkipVerify: true}
+		}
+	}
+}
+
+func WithUser(user string) Option {
+	if user == "" {
+		panic("user is empty")
```
Comment (BinLiam) [inline pkg/dsn_util/postgresql.go:160]: Oh, that's too bad.

File: pkg/dsn_util/postgresql.go [hunk 4]
```diff
@@ -0,0 +1,452 @@
...
+
+func WithPassword(password string) Option {
+	return func(cfg *Config) {
+		cfg.Password = password
+	}
+}
+
+func WithDatabase(database string) Option {
+	if database == "" {
+		panic("database is empty")
```
Comment (BinLiam) [inline pkg/dsn_util/postgresql.go:175]: Oh, that's too bad.

File: pkg/dsn_util/postgresql.go [hunk 5]
```diff
@@ -0,0 +1,452 @@
...
+//	return func(cfg *Config) {
+//		cfg.ResetSessionFunc = fn
+//	}
+//}
+
+func WithDSN(dsn string) Option {
+	return func(cfg *Config) {
+		opts, err := parseDSN(dsn)
+		if err != nil {
+			panic(err)
```
Comment (BinLiam) [inline pkg/dsn_util/postgresql.go:233]: Oh, that's too bad.

Comment (monserrat101) [reply pkg/dsn_util/postgresql.go:233]: It's okay. There's a recovery at the entrance

Comment (kennethBo) [reply pkg/dsn_util/postgresql.go:233]: LGTM

**Pull request merged**